### ISSUE TITLE
Confusing behaviour with `perm` in a quantifier for quantified vs non-quantified fields

### ISSUE THREAD

**Issue by superaxander** [viperproject/silicon#868]
For this program Viper reports that there might be insufficient permission to access `i.element` when inhaling the quantifier at the bottom:

```rust
field element: Int

function id(x:Int): Int
{
  x
}

method foo()
{
  var a: Ref
  var b: Ref
  inhale acc(a.element, write)
  inhale acc(b.element, write)
  //inhale forall i: Ref :: {i.element} perm(i.element) > none ==> id(i.element) == id(i.element)
  inhale forall i: Ref :: {id(i.element)} perm(i.element) > none ==> id(i.element) == id(i.element)
}
```

The commented out quantifier *can* be inhaled because its trigger causes the element field to be added to the `State.qpFields` set. Additionally, if there is only one field chunk for the `element` field the quantifier can also be inhaled. In the latter case while evaluating `id(i.element)` the path condition contains `(i == a ? W : Z) > Z` which obviously implies `i == a` making permission lookup succeed for `i.element`. However for the case with two field chunks the path condition becomes `(i == b ? W : Z) + (i == a ? W : Z) > Z` but matching on non-quantified field chunks allows only one field chunk to be consumed whereas we need permission from both here. 

Long story short this quantifier requires the field to be treated as a quantified field but this only happens for triggers where the expression is a field access. I'm guessing the best solution would be to detect more cases where a field should be added to the qpFields set. Perhaps looking for a `perm` expression in the quantifier body, but there are probably other ways to run into this same problem

**Comment by viper-admin** [comment]
Silicon in its default mode is generally incomplete in states with disjunctive aliasing (like ``i == a || i == b``), I think your problem is an instance of that. 
Making a resource a QP resource solves that, but the better way to solve the problem is to switch to ``--exhaleMode=1`` (formerly known as ``--enableMoreCompleteExhale``), which solves that problem. Or alternatively ``--exhaleMode=2``, which uses Silicon's default algorithm by default but falls back on the more complete mode when it encounters a failure.

**Comment by superaxander** [comment]
> Silicon in its default mode is generally incomplete in states with disjunctive aliasing (like `i == a || i == b`), I think your problem is an instance of that. Making a resource a QP resource solves that, but the better way to solve the problem is to switch to `--exhaleMode=1` (formerly known as `--enableMoreCompleteExhale`), which solves that problem. Or alternatively `--exhaleMode=2`, which uses Silicon's default algorithm by default but falls back on the more complete mode when it encounters a failure.

Ah thanks I forgot to try that, that works indeed
